localstack/services/apigateway/helpers.py

```diff
@@ -246,10 +246,25 @@ def get_resource_for_path(path, path_map):
         for match in matches:
             if match[0] == path:
                 return match
+            if test_path(path, match[0]):
+                return match
         raise Exception('Ambiguous API path %s - matches found: %s' % (path, matches))
     return matches[0]
 
 
+def test_path(path, apiPath):
+    api_paths = apiPath.split('/')
+    paths = path.split('/')
+    matched = False
+    reg_check = re.compile(r'\{(.*)\}')
+    if len(api_paths) != len(paths):
+        return False
+    for indx, part in enumerate(api_paths):
+        if reg_check.match(part) is None and part:
+            matched = part == paths[indx]
```

Comment:
Looks like we're overwriting `matched` in each iteration of the loop - i.e., a value of `matched = False` could be overwritten with `matched = True` in a subsequent iteration.

Can we please add another unit test with `path` as `'/foo/bar/baz'` and `api_path` containing an entry like `'/foo123/{param1}/baz'` - which should **not** match, but I believe it would match in the current implementation..?

Reply:
Yep makes sense - will at the test and fix